test/sys/test_socket.rs

```diff
@@ -2257,6 +2257,162 @@ pub fn test_recv_ipv6pktinfo() {
     }
 }
 
+#[cfg(linux_android)]
+#[test]
+pub fn test_tos_ipv4() {
+    use nix::sys::socket::sockopt::{IpTos, Ipv4RecvTos};
+    use nix::sys::socket::{bind, SockFlag, SockType, SockaddrIn};
+    use nix::sys::socket::{getsockname, setsockopt, socket};
+    use nix::sys::socket::{recvmsg, sendmsg, ControlMessageOwned, MsgFlags};
+    use std::io::{IoSlice, IoSliceMut};
+
+    let lo_ifaddr = loopback_address(AddressFamily::Inet);
+    let (_lo_name, lo) = match lo_ifaddr {
+        Some(ifaddr) => (
+            ifaddr.interface_name,
+            ifaddr.address.expect("Expect IPv4 address on interface"),
+        ),
+        None => return,
+    };
+    let receive = socket(
```

Comment:
Wouldn't it be easier to write the test to use `socketpair` instead of to create two sockets and connect them?

Reply:
I just copied this pattern from nearby tests. Might be better to leave it as is for the sake of consistency?